### PR TITLE
User Authentication , File and Folder structure

### PR DESCRIPTION
**Pull request by yaseerokino**
Hey Team,

In this PR the following was completed: 
- Setup and configure authentication and authorization for the front end;
- Setup and configured redux store.
- Setup folder structure and linters.

### CODE REVIEW

File: src/store/user.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    if (user && jwt) {
+      const signedUpUser = { ...user, token: jwt };
+      return dispatch(signupSuccess(signedUpUser));
+    }
+    if (errors) {
+      return dispatch(signupFailure());
+    }
+  } catch (e) {
+    // console
```
Comment (Mwapsam) [inline src/store/user.js:93]: I would suggest you remove this line if it does not save any purpose or complete it. Leaving it empty might cause linter errors.

`catch (e) {
    throw new Error(e);
  }`

**Review by Mwapsam — CHANGES_REQUESTED** [review]
Hey @yaseerokino 😃 
Congratulations on implementing user authentication using react and redux! 👏🏼 🎉

**HIGHLIGHTS**

- Made use of the redux toolkit
- Made of localStorage hash to maintain user session
- consumed the backend API efficiently by posting and fetching data from it
- Implemented React routing for navigation
- Clean and professional code

**_Great job done but please fix the linter errors before merging to the dev branch, this will help us maintain the clean code base_**

**Comment by yaseerokino** [comment]
The linter errors are there because of the empty files.

**Comment by Mwapsam** [comment]
Okay then I nothing else to say, please go ahead and merge it so that everyone can start working on it

**Review by Mwapsam — APPROVED** [review]

**Pull request merged**